xdsl/parser.py

```diff
@@ -456,17 +456,45 @@ def _find_token_end(self, start: int | None = None) -> int:
         """
         Find the point (optionally starting from start) where the token ends
         """
+        # get the start of the token
         i = self.next_pos() if start is None else start
-        # Search for literal breaks
+
+        # we only look 100 characters ahead when searching for a token.
+        # this is a large file optimization. Since we are searching for each break
+        # symbol independently, we sometimes look very far into the file before
+        # we find it. By limiting our lookahead to a few characters, we can cover
+        # most cases, and only have to fall back to a slow scan in
+        # relatively few cases.
+        lookahead = 100
+        # the end of our token
+        end_of_token = i + lookahead
+
+        # now do a scan for each thing we break on
         for part in self.break_on:
```

Comment:
I am not very familiar with the parsing infrastructure, but it feels like you could maybe have break_on ordered by position in the file, so this would only be a simple access, instead of having to scan through all of them.

Maybe this is bs, if so, just ignore it :)